hnn_core/extracellular.py

```diff
@@ -28,6 +28,75 @@
 from .externals.mne import _validate_type, _check_option
 
 
+def _calculate_csd2d(lfp_data, ch_axis=0, delta=1):
+    """Current source density (CSD) estimation
+
+    The three-point finite-difference approximation of the
+    second spatial derivative for computing 1-dimensional CSD
+    with border electrode interpolation
+    csd[electrode] = -(LFP[electrode - 1] - 2*LFP[electrode] +
+                       LFP[electrode + 1]) / spacing ** 2
+
+    Parameters
+    ----------
+    lfp_data : channels x times array
+        LFP data.
+    ch_axis : int
+        Axis of electrode array, default is zero.
+    delta : int
+        Spacing between channels, scales the CSD.
+
+    Returns
+    -------
+    csd2d : channels x times array
+        the 2nd derivative current source density estimate (csd2d)
+    """
+    csd2d = -np.diff(np.diff(lfp_data, axis=0), axis=0) / delta ** 2
+    bottom_border = csd2d[-1, :] * 2 - csd2d[-2, :]
+    top_border = csd2d[0, :] * 2 - csd2d[1, :]
+    csd2d = np.concatenate([np.expand_dims(top_border, axis=ch_axis),
+                            csd2d, np.expand_dims(bottom_border,
+                                                  axis=ch_axis)], axis=ch_axis)
+    return csd2d
+
+
+def _get_laminar_z_coords(electrode_positions):
+    """Get equispaced, colinear electrode coordinates along z-axis.
+
+    Parameters
+    ----------
+    electrode_positions : list of tuple
+        The (x, y, z) coordinates (in um) of the extracellular electrodes.
+
+    Returns
+    -------
+    z_coordinates : array, shape (n_contacts,)
+        Z-coordinates of the electrode contacts.
+    z_delta : float
+        Magnitude of change in the z-direction.
+    """
+    n_contacts = np.array(electrode_positions).shape[0]
+    if n_contacts < 2:
+        raise ValueError(
+            'Electrode array positions must contain more than 1 contact to be '
```

Comment:
more than 1 or more than 2? The formula seems to require at least 2 ...

Reply:
Correct, more than 1 (or at least 2).